test/externalTests/gnosis.sh

```diff
@@ -106,6 +106,10 @@ function gnosis_safe_test
     # pulls @ethersproject/contracts@5.6.1 (latest). Force 5.6.0 to avoid errors due to having two copies.
     npm install @ethersproject/contracts@5.6.0
 
+    # 2.1.1 started causing failures in safe-contracts external tests after a contract address check was introduced
+    # in https://github.com/NomicFoundation/hardhat/pull/2916, and so to avoid errors, the package is now pegged.
+    npm install @nomiclabs/hardhat-ethers@2.1.0
```

Comment:
Please also add something like
```
 # TODO: Remove when <issue/PR link> is resolved.
```
I.e. this is really something that Gnosis has to fix and we should have an easy link where we can check if it's fixed and then remove it.